medusa/server/api/v2/base.py

```diff
@@ -81,72 +114,21 @@ def _check_authentication(self):
         else:
             return self._unauthorized('Invalid token.')
 
-    def async_call(self, name, *args, **kwargs):
-        """Call the actual HTTP method, if available."""
-        try:
-            method = getattr(self, 'http_' + name)
-        except AttributeError:
-            raise HTTPError(405)
-
-        def blocking_call():
-            result = self._check_authentication()
-            return method(*args, **kwargs) if result is None else result
-
-        return IOLoop.current().run_in_executor(executor, blocking_call)
-
-    @coroutine
-    def head(self, *args, **kwargs):
-        """HEAD HTTP method."""
-        content = self.async_call('head', *args, **kwargs)
-        if content is not None:
-            content = yield content
-        if not self._finished:
-            self.finish(content)
-
-    @coroutine
-    def get(self, *args, **kwargs):
-        """GET HTTP method."""
-        content = self.async_call('get', *args, **kwargs)
-        if content is not None:
-            content = yield content
-        if not self._finished:
-            self.finish(content)
-
-    @coroutine
-    def post(self, *args, **kwargs):
-        """POST HTTP method."""
-        content = self.async_call('post', *args, **kwargs)
-        if content is not None:
-            content = yield content
-        if not self._finished:
-            self.finish(content)
-
-    @coroutine
-    def delete(self, *args, **kwargs):
-        """DELETE HTTP method."""
-        content = self.async_call('delete', *args, **kwargs)
-        if content is not None:
-            content = yield content
-        if not self._finished:
-            self.finish(content)
+    def initialize(self):
+        """
+        Override the request method to use the async dispatcher.
 
-    @coroutine
-    def patch(self, *args, **kwargs):
-        """PATCH HTTP method."""
-        content = self.async_call('patch', *args, **kwargs)
-        if content is not None:
-            content = yield content
-        if not self._finished:
-            self.finish(content)
+        This function is called for each request.
+        """
+        name = self.request.method.lower()
+        # Wrap the original method with the code needed to run it asynchronously
+        method = make_async(self, getattr(self, name))
```

Comment:
By the way, this means `OPTIONS` will also be asynchronous.
It doesn't mean much, I'm just saying it because currently it is synchronous.